schema/snapcraft.yaml

```diff
@@ -276,6 +276,9 @@ properties:
           stage-packages:
             $ref: "#/definitions/grammar-array"
             default: [] # For some reason this doesn't work if in the ref
+          build-snaps:
+            $ref: "#/definitions/grammar-array"
+            default: [] # For some reason this doesn't work if in the ref
```

Comment:
When I initially wrote the grammar back in the day I wasn't sure what was happening, but I think I know now so we can update the comment: a jsonschema ref isn't expanded in any fancy way by the YAML (which is why I'd like to use a recursive YAML anchor for this, but jsonschema causes things to explode that way since it wants to walk the whole thing). This means that we can't actually access the chunk of YAML representing that bit of the schema very easily in code. Like we do [here](https://github.com/snapcore/snapcraft/blob/master/snapcraft/internal/pluginhandler/__init__.py#L628) to expand defaults. So this is kind of an ugly hack to allow specifying the defaults to continue actually resulting in defaults.